### ISSUE TITLE
2018/08/26 deeplab v3 covert numpy generator to auto-update tensor

### ISSUE THREAD

**Issue by yzbx** [ISCAS007/torchseg#14]
# evaluating tnesors
TensorFlow can't directly evaluate tf.Tensors defined inside functions or inside control flow constructs. If a tf.Tensor depends on a value from a queue, evaluating the tf.Tensor will only work once something has been enqueued; otherwise, evaluating it will hang. When working with queues, remember to call tf.train.start_queue_runners before evaluating any tf.Tensors.

# Initializing variables
Before you can use a variable, it must be initialized. If you are programming in the low-level TensorFlow API (that is, you are explicitly creating your own graphs and sessions), you must explicitly initialize the variables. Most high-level frameworks such as tf.contrib.slim, tf.estimator.Estimator and Keras automatically initialize variables for you before training a model.

**Comment by yzbx** [comment]
# slim
- [awesome slim](https://github.com/xssc/awesome-slim)

**Comment by yzbx** [comment]
# experiment
- resnet101 + data augmentation: train miou = 0.75, val miou = 0.47
```
python test/pspnet_test.py --batch_size=8 --net_name=pspnet --augmentation=True --learning_rate=0.01 --optimizer=sgd --backbone_name=resnet101 --midnet_scale=10 --upsample_type=bilinear --backbone_pretrained=True
```

- resnet101 + no data augmentation: train miou = 0.75, val miou = 0.47
```
python test/pspnet_test.py --batch_size=8 --net_name=pspnet --augmentation=False --learning_rate=0.01 --optimizer=sgd --backbone_name=resnet101 --midnet_scale=10 --upsample_type=bilinear --backbone_pretrained=True
```

- more epoch for large learning rate (default epoch=100)
```python test/pspnet_test.py --batch_size=8 --net_name=pspnet --augmentation=True --learning_rate=0.01 --optimizer=sgd --backbone_name=resnet101 --midnet_scale=15 --upsample_type=bilinear --backbone_pretrained=True --n_epoch=200```

**Comment by yzbx** [comment]
# convert caffe weight to pytorch
caffe weight == converted pytorch weight

dataset | caffe weight | pytorch weight | miou
:-: | :-: | :-: | :-:
voc | use | - | 0.08-0.09
voc | - | use | 0.08-0.09
cityscapes | use | - | 0.26-0.28
cityscapes | - | use | 0.26-0.28

**Comment by yzbx** [comment]
# pspnet-pytorch + default dataloader
```                          
  "FreqW Acc": 0.9451615135203637,
  "Overall Acc": 0.9712679557972358,
  "Mean Acc": 0.9109202561712676,
  "Mean IoU": 0.8655772918265227
  0: 0.96752136444022852, 
  1: 0.91617778918957049, 
  2: 0.67929072462895501, 
  3: 0.91331709115100734, 
  4: 0.81547357469744319, 
  5: 0.86104017649846865, 
  6: 0.96270114163429799, 
  7: 0.92712886355212165, 
  8: 0.93012528299837416,
  9: 0.65205186713059538, 
  10: 0.93994314552086222, 
  11: 0.80752690197961907, 
  12: 0.91038228746719374, 
  13: 0.91186950599472594, 
  14: 0.8980326761640921, 
  15: 0.89131221738346722, 
  16: 0.77336775974055583, 
  17: 0.92077758318739056, 
  18: 0.80442515788599522, 
  19: 0.92065957766827866, 
  20: 0.77399843944373825
```
